experiment/builder_runner.py

```diff
@@ -308,20 +308,36 @@ def build_and_run(self, generated_project: str, target_path: str,
     project_target_name = os.path.basename(self.benchmark.target_path)
     benchmark_log_path = self.work_dirs.build_logs_target(
         benchmark_target_name, iteration)
+    jcc_errlog_path = self.work_dirs.jcc_errlog_target(benchmark_target_name,
+                                                       iteration)
     build_result.succeeded = self.build_target_local(generated_project,
                                                      benchmark_log_path)
     # Copy err.log into work dir.
     try:
       shutil.copyfile(
           os.path.join(get_build_artifact_dir(generated_project, "workspace"),
-                       'err.log'),
-          self.work_dirs.error_logs_target(benchmark_target_name, iteration))
+                       'err.log'), jcc_errlog_path)
     except FileNotFoundError as e:
       logging.error('Cannot get err.log for %s: %s', generated_project, e)
+      # Touch err.log in results folder to avoid FileNotFoundError when
+      # extracting errors.
+      open(jcc_errlog_path, 'x')
```

Comment:
> Maybe add a magic string in err.log when err.log does not exist?

This can work but not preferred.
Intuitively, if `err.log` is not generated, then it should not exist.

>  creating an empty file is currently consistent with the behaviour of getting the build log and run log

Where do we create empty build log and run logs when they do not exist?

> Might help prevent increasing the complexity of build_and_run() workflow?

Not sure if this relates, but I suppose this will only add two lines in `code_fixer`?
>> Could we check os.path.isfile() at the beginning extract_error_message() instead?

Reply:
> Where do we create empty build log and run logs?

It's in cloud builder, we open the local file object before checking if the file blob on cloud exists
https://github.com/google/oss-fuzz-gen/blob/main/experiment/builder_runner.py#L645

Reply:
Ah, I see; let's keep it, then.
Thanks!